### PR TITLE
REST Webserver

### PR DESCRIPTION
**Pull request by t-mon**
prepared tcp server and mock server for webserver
add TransportInterface
add basic structure of webserver

### CODE REVIEW

File: server/transportinterface.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+#include <QObject>
+
+namespace guhserver {
+
+class TransportInterface : public QObject
+{
+    Q_OBJECT
+public:
+    explicit TransportInterface(QObject *parent = 0);
```
Comment (mzanetti) [inline server/transportinterface.h]: virtual destructor missing!

File: server/tcpserver.h [hunk 2]
```diff
@@ -51,12 +53,12 @@ class TcpServer : public QObject
     QStringList m_ipVersions;
 
     void reloadNetworkInterfaces();
+    void validateMessage(const QUuid &clientId, const QByteArray &data);
```
Comment (mzanetti) [inline server/tcpserver.h]: add "override" keywords

File: server/tcpserver.h [hunk 3]
```diff
@@ -29,16 +29,18 @@
...
+class TcpServer : public TransportInterface
 {
     Q_OBJECT
 public:
     explicit TcpServer(QObject *parent = 0);
     
-    void sendData(const QUuid &clientId, const QByteArray &data);
-    void sendData(const QList<QUuid> &clients, const QByteArray &data);
+    void sendData(const QUuid &clientId, const QVariantMap &data);
+    void sendData(const QList<QUuid> &clients, const QVariantMap &data);
```
Comment (mzanetti) [inline server/tcpserver.h]: "override" for both

File: tests/auto/mocktcpserver.h [hunk 4]
```diff
@@ -25,17 +25,21 @@
...
 {
     Q_OBJECT
 public:
     explicit MockTcpServer(QObject *parent = 0);
     ~MockTcpServer();
 
-    void sendData(const QUuid &clientId, const QByteArray &data);
-    void sendData(const QList<QUuid> &clients, const QByteArray &data);
+    void sendData(const QUuid &clientId, const QVariantMap &data);
+    void sendData(const QList<QUuid> &clients, const QVariantMap &data);
```
Comment (mzanetti) [inline tests/auto/mocktcpserver.h]: override

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/303/
Test PASSed.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/304/
Test PASSed.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/305/
Test PASSed.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/306/
Test PASSed.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/307/
Test PASSed.

**Comment by t-mon** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/308/
Test FAILed.

**Comment by t-mon** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/309/
Test FAILed.

**Comment by t-mon** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/310/
Test FAILed.

**Comment by t-mon** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/311/
Test FAILed.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/312/
Test PASSed.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/313/
Test PASSed.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/314/
Test PASSed.

**Comment by t-mon** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/315/
Test FAILed.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/316/
Test PASSed.

**Comment by t-mon** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://guh.guru:8080/job/guh-pull-request-builder/317/
Test PASSed.

**Pull request merged**